app/controllers/katello/api/v2/subscriptions_controller.rb

```diff
@@ -240,6 +242,25 @@ def index_organization
 
       return subscriptions
     end
+ 
+    def consumed_subscriptions(consumer, organization = current_organization)
+      entitlements = consumer.consumed_entitlements
+      entitlements.delete_if do |entitlement|
+        pool = Pool.find_pool(entitlement.poolId)
+        !pool.has_provider?(organization)
+      end
+    end
+
+    def available_subscriptions(cp_pools, organization = current_organization)
+      if cp_pools
+        pools = cp_pools.collect { |cp_pool| ::Katello::Pool.find_pool(cp_pool['id']) }
+        pools.compact!
+        pools.delete_if { |pool| !pool.has_provider?(organization) }
+      else
+        pools = []
```

Comment:
This doesn't need to be set because you are just returning from here. And you could restructure to:

```
def available_subscriptions(cp_pools, organization = current_organization)
  return [] unless cp_pools
  pools = cp_pools.collect { |cp_pool| ::Katello::Pool.find_pool(cp_pool['id']) }
  pools.compact!
  pools.select { |pool| pool.has_provider?(organization) }
end
```

Could this be re-written above, given that there is now information stored in our database? or as a scope?